linkerd/metrics/src/counter.rs

```diff
@@ -14,41 +17,56 @@ use std::sync::atomic::{AtomicU64, Ordering};
 /// [`rate()`]: https://prometheus.io/docs/prometheus/latest/querying/functions/#rate()
 /// [`irate()`]: https://prometheus.io/docs/prometheus/latest/querying/functions/#irate()
 /// [`resets()`]: https://prometheus.io/docs/prometheus/latest/querying/functions/#resets
-#[derive(Debug, Default)]
-pub struct Counter(AtomicU64);
+#[derive(Debug)]
+pub struct Counter<F = ()>(AtomicU64, std::marker::PhantomData<F>);
 
 // ===== impl Counter =====
 
-impl Counter {
+impl Default for Counter {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl<F: Factor> Counter<F> {
+    pub fn new() -> Self {
+        Self(AtomicU64::default(), std::marker::PhantomData)
+    }
+
     pub fn incr(&self) {
-        self.add(1)
+        self.add(1u64)
```

Comment:
huh, i'm surprised the explicit type is necessary here...

Reply:
Cruft from an intermediate state where `add` took `impl Into<u64>` -- no longer the case so removed the type.